sklearn/metrics/pairwise.py

```diff
@@ -32,6 +33,15 @@
 from .pairwise_fast import _chi2_kernel_fast, _sparse_manhattan
 
 
+# Get mask for missing values
+def _get_mask(X, value_to_mask):
```

Comment:
should we move this to utils?

Reply:
Yes, please

Reply:
Please do move this to utils

Reply:
Refactoring `_get_mask` to `utils` would causes changes unrelated to `KNNImputer` and `nan_euclidean_distances`. I can open another PR for this refactor.

Reply:
On second thought, I'll refactor it here. The addition of `KNNImputer` motivates the refactor.